Add tests (Starting to anyway)

escardin opened the pull request.

chenkie commented:
@escardin it looks like there are some conflicts since I merged other stuff. Can you check them and I'll get this in? :)

escardin commented:
I cleaned it up and added a few more tests.

I have a question about the intent of offsetSeconds in jwtHelper.isTokenExpired. Is it meant as a grace period to deal with clock drift? Right now it's expiring tokens earlier than expected (see the commented out tests).

chenkie commented:
Sorry this took a while--thanks very much for these tests :)

I'm going to check about `offsetSeconds`. That particular function was borrowed from another of our libraries.
